### ISSUE TITLE
We should try Wang/Srinivasan's method with our data

### ISSUE THREAD

**Issue by moorepants** [csu-hmc/gait-control-direct-id-paper#48]
The Wang/Srinivasan paper found that foot placement is well predicted by the mid stance hip state. They used unperturbed walking. This should be fairly easy to reproduce with our perturbed data to see if there is similar or stronger results. Also it would be interesting to use a larger set of variables than them and then reduce to see if we get the same essential important predictors.

**Comment by moorepants** [comment]
From Andy:

> Andy’s hypothesis:  That COM velocity is not the best variable for predicting
foot placement.  Maybe some combination of terms that includes angular momentum
about the stance foot.

**Comment by moorepants** [comment]
Also from Andy:
> 1) You have marker positions vs time.  You only correlate those at one time with those
at later times.  Manoj’s thing is a simple example of this.  In this case, some correlations
follow from the laws of mechanics, and some don’t.   So you have to be thoughtful,
but that doesn’t mean it is not interesting.

**Comment by tvdbogert** [comment]
Yes, you have to be very careful with these things.  I did not have a good feeling about Manoj's results, they may be easily explained by the laws of mechanics.  This may be again just the basic problem of closed loop system identification, that you don't know which part of the system you are really identifying.   

Relationship between COM and foot placement.  Clearly there is potential for balance control, but identifying a control law from data is scary. Foot position and COM position are so strongly correlated through the kinematics.  If you do a larger step (for whatever reason), the COM is going to be more forward than usual, when you're halfway into the step.  This is kinematics, not control. If you used the COM at the beginning of the step, you might avoid this correlation.

Not that there's anything necessarily wrong with a control law that reflects kinematics or dynamics of the system being controlled, it may just work even if it is the result from erroneous system identification.  Especially in prosthetics and exoskeletons where there is additional human control to help out.

**Comment by moorepants** [comment]
> they may be easily explained by the laws of mechanics

Yes, this is a pitfall. For example, if you collect data from a self-stable simple walking machine and run this analysis you may get a similar result but by definition it doesn't have control, so all you identify is the open loop relationship. In this case though, we know the system isn't self-stable so there could be a chance it is picking up control relationships, but it is difficult to know what portion of the loop you are identifying.

> If you used the COM at the beginning of the step, you might avoid this correlation.

It would be nice to plot the model's ability to predict the foot placement as a function of the state before foot fall, so that you can see if and how much it diminishes the further you move back in time from the foot fall. The further back you go the less kinematic correlation you have.

> Not that there's anything necessarily wrong with a control law that reflects kinematics or dynamics of the system being controlled

There are certainly controlled systems that act only on position control with position sensed kinematics, but the controller has to be very ideal, i.e. be able to provide any force/torque needed to "instantaneously" move to a position. The bicycle could even potentially described that way because it takes so little torque to rotate the handlebars that maybe the person is in fact doing position control. I can imagine that it would be possible to create a very simple, lightweight walking robot that could act with a position control law like they identified here. But something like that seems far from a real human system or a complex robot.

Andy suggested that since I have potentially "better" data, i.e. more potential variables to chose from, that it would be interesting in its own right to see if we can 1) reproduce Manoj's result and 2) can we get a better result by utilizing more information that we have. I thought this same thing when I first read Manoj's paper. But I'm not sure it will shed any light or enhance the current claims and path of the paper, so it may not be worth spending time on for this paper. The only thing that may need to be explored with respect to our work is having a delay in our controller, see #3, which Manoj's study has made me think about that more.

**Comment by tvdbogert** [comment]
Let's stick with the gain-scheduled PD control and not do foot placement just yet.  There is ample justification for using gain-scheduled PD control in prosthetics and exoskeletons.

I commented on #3.

**Comment by moorepants** [comment]
Ok, I'll keep this in mind for when I have all that spare time in the future :)
